gluten-core/src/main/scala/org/apache/spark/softaffinity/SoftAffinityUtil.scala

```diff
@@ -23,42 +23,40 @@ import io.glutenproject.utils.LogLevelUtil
 
 import org.apache.spark.internal.Logging
 import org.apache.spark.scheduler.ExecutorCacheTaskLocation
-import org.apache.spark.sql.execution.datasources.FilePartition
 
 object SoftAffinityUtil extends LogLevelUtil with Logging {
 
   private lazy val softAffinityLogLevel = GlutenConfig.getConf.softAffinityLogLevel
 
   /** Get the locations by SoftAffinityManager */
-  def getFilePartitionLocations(filePartition: FilePartition): Array[String] = {
-    // Get the original preferred locations
-    val expectedTargets = filePartition.preferredLocations()
-
+  def getFilePartitionLocations(
+      filePaths: Array[String],
+      preferredLocations: Array[String]): Array[String] = {
```

Comment:
can you explain this definition a little?

Reply:
Previously, the input of this method was `FilePartition`, but the `InputPartitions` for lake formats are custom-defined. This method primarily processes `FilePartition.files.filePath` and `preferredLocations`. Modifying the input type of this method would allow the lake formats to directly reuse this interface.